format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSourceConfig.java

```diff
@@ -133,6 +140,17 @@ public void validate(FailureCollector collector) {
       collector.addFailure(e.getMessage(), null).withConfigProperty(NAME_SCHEMA).withStacktrace(e.getStackTrace());
     }
 
+    if (fileEncoding != null && !fileEncoding.isEmpty()) {
+      fileEncoding = cleanFileEncodingName(fileEncoding);
```

Comment:
it's better not to mutate any fields, as it could lead to weird bugs if getFileEncoding() is called before validate().

Instead, it seems like we should change the getter to clean the encoding name before returning it. This incurs the extra cost of cleaning per get call, but it shouldn't be a big deal.